File: application/app/domain/model/filters/BlacklistHosts.scala
package domain.model.filters

import common.validation.constraints.TraversableSize
import domain.model.{Filter, SetValue}
import javax.validation.constraints.NotNull

import scala.annotation.meta.field

case class BlacklistHosts(
                           @(NotNull @field) @(TraversableSize @field)(max = 255) blacklist: Set[String]
                         ) extends Filter {
  override val name: String = Filter.BLACKLIST_HOSTS
  override val value: SetValue = SetValue(blacklist)
  override val filter: Map[String, String] => Boolean = headers => !headers.contains("host") || !blacklist.contains(headers("host"))
}

File: ws-gateway-domain/src/main/java/com/cosmin/wsgateway/domain/model/filters/BlacklistHosts.java
package com.cosmin.wsgateway.domain.model.filters;

import com.cosmin.wsgateway.domain.model.Filter;

import lombok.Value;

import java.util.Map;
import java.util.Set;

@Value
public class BlacklistHosts implements Filter<Set<String>> {
    private final Set<String> hosts;

    @Override
    public String name() {
        return "blacklistHosts";
    }

    @Override
    public Set<String> value() {
        return hosts;
    }

    @Override
    public boolean isAllowed(Map<String, String> requestHeaders) {
        return !requestHeaders.containsKey("Host") || !hosts.contains(requestHeaders.get("Host"));
    }
}
